src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -95,21 +118,44 @@
     public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidContainerItemComponent fluidContainer,
                                        ItemComponent itemComponent) {
         if (fluidContainer.fluidType == null || fluidContainer.volume < fluidContainer.maxVolume) {
-            getLiquidInReach(event.getInstigatorLocation(), event.getDirection(), 3).ifPresent(block -> {
+            getLiquidInReach(event.getInstigatorLocation(), event.getDirection(), 3).ifPresent(pos -> {
                 EntityRef owner = item.getOwner();
                 final EntityRef removedItem = inventoryManager.removeItem(owner, event.getInstigator(), item, false, 1);
                 //TODO: replace with better fluid handling, maybe by new CoreFluids module
-                if (removedItem != null && block.isWater()) {
+                if (removedItem != null && worldProvider.getBlock(pos).isWater()) {
+                    byte liquidData = (byte) worldProvider.getExtraData(flowIx, pos);
```

Comment:
I don't think that works as a static method: it needs access to the `ExtraBlockDataManager` to get the index. It would be possible to get the `LiquidFlowSystem` through the `ComponentSystemManager`, and have this as a non-static method in there (without the `WorldProvider` argument in that case, as it's already accessible). That wouldn't make the `FluidAuthoritySystem` so much neater though.